vortex-array/src/arrow/compute/to_arrow/canonical.rs

```diff
@@ -48,6 +48,22 @@ impl Kernel for ToArrowCanonical {
             .map(Ok)
             .unwrap_or_else(|| array.dtype().to_arrow_dtype())?;
 
+        if matches!(arrow_type, DataType::Decimal128(..)) {
+            let arrow_array = to_arrow_decimal128(array.to_decimal()?)?;
```

Comment:
I don't understand why this is a special case? You're still calling `array.to_decimal()` which effectively canonicalizes no?

Reply:
Not sure it's a huge difference honestly. We can work on it if it does end up being important.